core/src/test/java/com/graphhopper/routing/ch/NodeBasedNodeContractorTest.java

```diff
@@ -288,6 +287,65 @@ public void testContractNode_directed_withWitness() {
         checkNoShortcuts();
     }
 
+
+    @Test
+    public void testNodeContraction_shortcutWeightRounding() {
+        // 0 ------------> 4
+        //  \             /
+        //   1 --> 2 --> 3 
+        double[] distances = {4.019, 1.006, 1.004, 1.006, 1.004};
+        graph.edge(0, 4, distances[0], false);
+        EdgeIteratorState edge1 = graph.edge(0, 1, distances[1], false);
+        EdgeIteratorState edge2 = graph.edge(1, 2, distances[2], false);
+        EdgeIteratorState edge3 = graph.edge(2, 3, distances[3], false);
+        EdgeIteratorState edge4 = graph.edge(3, 4, distances[4], false);
+        graph.freeze();
+        setMaxLevelOnAllNodes();
+
+        // make sure that distances do not get changed in storage (might get truncated)
```

Comment:
When edges get stored the distance is also truncated, for example 1.006 can become 1.005. I tried choosing values where this does not happen because this is for another issue.